torchdynamo/symbolic_convert.py

```diff
@@ -553,6 +554,39 @@ def GET_ITER(self, inst):
     def CALL_FUNCTION(self, inst):
         args = self.popn(inst.argval)
         fn = self.pop()
+
+        # Hack to handle special case in some legacy code.
+        # Convert x.new(torch.Size) into torch.empty(tuple),
+        # as torch.Tensor.new acts differently with a Size input versus a tuple input.
+        if (
+            isinstance(fn, GetAttrVariable)
+            and isinstance(fn.obj, TensorVariable)
+            and fn.name == "new"
+            and len(args) == 1
+            and isinstance(args[0], SizeVariable)
+            and not config.dynamic_shapes
```

Comment:
I'd suggest also checking that `fn.obj.dtype` / `fn.obj.device` are both non-None.  This will currently only happen if you disable dynamic_propagation.